packages/material-ui/src/Autocomplete/Autocomplete.js

```diff
@@ -205,20 +204,26 @@ export const styles = (theme) => ({
     [theme.breakpoints.up('sm')]: {
       minHeight: 'auto',
     },
-    '&[aria-selected="true"]': {
-      backgroundColor: theme.palette.action.selected,
-    },
-    '&[data-focus="true"]': {
+    '&$focused': {
       backgroundColor: theme.palette.action.hover,
     },
-    '&:active': {
+    '&$selected': {
       backgroundColor: theme.palette.action.selected,
+      '&$focused': {
+        backgroundColor: alpha(
```

Comment:
This change is related to #18136. I think that we should have a different pull request for it. It would deliver value, on its own :).

Reply:
@oliviertassinari  I changed the above classes names to unique names `optionFocused, optionSelected, optionDisabled` because [official doc](https://material-ui.com/api/autocomplete/#autocomplete-api) says `focused is Pseudo-class applied to the root element if focused`. Thanks for your review. :)

Reply:
In terms of #18136, I will send a different PR. :)